src/HttpHandler.h

```diff
@@ -0,0 +1,53 @@
+// Copyright (c) 2021 Pei-Ru Wang, Cheryl Huang, Yuan Zhou. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+/**
+ * Project MyHttpServer
+ * @author Pei-Ru Wang, Cheryl Huang, Yuan Zhou
+ * @version 0.0.1
+ */
+
+#ifndef _HTTPHANDLER_H
+#define _HTTPHANDLER_H
+
+#include <iostream>
+#include <string>
+#include <unordered_map>
+
+#include <boost/asio.hpp>
+#include <boost/bind.hpp>
+#include <boost/enable_shared_from_this.hpp>
+#include <boost/lexical_cast.hpp>
+#include <boost/shared_ptr.hpp>
+
+#include "HttpPacket.h"
+
+// Basic http handler class
+class HttpHandler {
+public:
+  // Initialize a basic http handler
+  HttpHandler(boost::asio::io_service *io_service);
+
+  // Avoid copying
+  HttpHandler(const HttpHandler &) = delete;
+  HttpHandler &operator=(const HttpHandler &) = delete;
+
+  // Avoid copying
+  HttpHandler(const HttpHandler &) = delete;
+  HttpHandler &operator=(const HttpHandler &) = delete;
+
+public:
+  // Return the socket instance
+  inline boost::asio::ip::tcp::socket *GetSocket() { return &socket; };
```

Comment:
inline is not needed. When you implement the function in the class definition, "inline" is implied.
https://en.wikipedia.org/wiki/Inline_function#C++
"The inline qualifier is automatically added to a function defined as part of a class definition."